### ISSUE TITLE
Nullable properties are optional at the same time

### ISSUE THREAD

**Issue by yss14** [victorgarciaesgi/simple-graphql-to-typescript#14]
Hi, it's me again, still enjoying your library :)

I just stumbled upon the following thing:
For fields in the graphql schema which are marked as nullable, it generates the interface properties in the following way: `someField?: Maybe<string>`
Thus, the property value can be either a string, `undefined`, or `null`. But from my understanding, graphql doesn't know `undefined`, just `null`. So if I query a field which value is not set, I receive `null`.
The only scenario where an optional interface property would make sense are types which are used in mutation or query arguments, because there an optional field can be omitted. 

So in the end, this results in the following questions:

1. Is there any reasoning or necessary to mark nullable fields also as optional in typescript?
2. Would it be possible to distinguish between types used as arguments and types which are returned by queries/mutations?
3. Would it be possible (maybe via a cli flag) to omit the "optional", or would this break too much code on your side which rely on these optional properties?

**Comment by victorgarciaesgi** [comment]
Hi! I'm really happy it's useful for someone else than me :D 

I'm aware of this problem, and I need to resolve it.
The current output is `someField?: Maybe<string>` to work with GraphQL input types.
Imagine you have a GraphQL type 
```graphql
input Signup {
   username: String
   email: String!
}
```

The generated output will be

```ts
interface Signup {
    username?: Maybe<string>;
    email: string;
}
```

So when I have a function `signup(data: Signup)`, I can call it like this `signup({email: 'test@test.com'})`

But if I generated 

```ts
interface Signup {
    username:Maybe<string>;
    email: string;
}
```

So when I have a function `signup(data: Signup)`, I would have call it like this `signup({email: 'test@test.com', username: null})`, otherwise Typescript would throw an error.

So I need to refactor my generators functions so INPUTS and OBJECT types have different output.

I will keep you updated!

**Comment by yss14** [comment]
Thanks for your fast response!

> So I need to refactor my generators functions so INPUTS and OBJECT types have different output.

If it is possible to distinguish between these two usages, that would be great!
Looking forward to here from you :)

**Comment by victorgarciaesgi** [comment]
@yss14 Ok that was relatively easy to do. So there is no need for another flag, now the generated output will be different depending of the usage!

For an interface (exemple):

```ts
export interface User {
  uuid: string;
  email: Maybe<string>;
  firstName: string;
  lastName: string;
  userName: string;
  description: string;
  pictureUrl: Maybe<string>;
  phone: Maybe<string>;
  gender: UserGenders;
  status: UserStatus;
}
```

For an input:
```graphql
export interface updateProfileArgs {
  firstName?: string;
  lastName?: string;
  userName?: string;
  email?: string;
}
```

Does this seems okay to you? I will also benefit from it so thanks for re-pointing me this problem :D

**Comment by victorgarciaesgi** [comment]
Available on 0.7.8!

**Comment by yss14** [comment]
Works perfectly. Thank you very much for your super fast support! :)
